builder/dockerfile/internals_windows.go

```diff
@@ -0,0 +1,392 @@
+	}
+
+	defer windows.RegCloseKey(usersKey)
+
+	err = changeKeySecurity(accountKey, "Groups")
+	if err != nil {
+		return "", err
+	}
+
+	err = windows.RegOpenKeyEx(accountKey, windows.StringToUTF16Ptr("Groups"), 0, windows.KEY_READ, &groupsKey)
+	if err != nil {
+		return "", err
+	}
+
+	defer windows.RegCloseKey(groupsKey)
+
+	err = changeKeySecurity(usersKey, "Names")
+	if err != nil {
+		return "", err
+	}
+
+	err = changeKeySecurity(groupsKey, "Names")
+	if err != nil {
+		return "", err
+	}
+
+	err = windows.RegOpenKeyEx(usersKey, windows.StringToUTF16Ptr("Names"), 0, windows.KEY_READ, &userNamesKey)
+	if err != nil {
+		return "", err
+	}
+
+	defer windows.RegCloseKey(userNamesKey)
+
+	userSid, locatedUserAccount, err := lookupNTUser(userNamesKey, accountName, computerSid)
+	if locatedUserAccount != false {
+		return userSid, nil
+	}
+
+	err = windows.RegOpenKeyEx(groupsKey, windows.StringToUTF16Ptr("Names"), 0, windows.KEY_READ, &groupNamesKey)
+	if err != nil {
+		return "", err
+	}
+
+	defer windows.RegCloseKey(groupNamesKey)
+
+	groupSid, locatedGroupAccount, err := lookupNTGroup(groupNamesKey, accountName, computerSid)
+	if locatedGroupAccount != false {
+		return groupSid, nil
+	}
+
+	return "", system.ERROR_NO_SUCH_USER
+}
+
+func lookupNTUser(userNamesKey windows.Handle, userStr string, computerSid string) (string, bool, error) {
+
+	var subKeyCount uint32
+	var maxSubKeyLen uint32
+	var accountName string
+
+	err := windows.RegQueryInfoKey(userNamesKey, nil, nil, nil, &subKeyCount, &maxSubKeyLen, nil, nil, nil, nil, nil, nil)
+	if err != nil {
+		return "", false, err
+	}
+
+	accountLocated := false
+	accountBuffer := make([]uint16, maxSubKeyLen+1)
+	maximumAccountBufferLength := uint32(len(accountBuffer))
+	accountBufferLength := maximumAccountBufferLength
+	accountSid := ""
+
+	for index := uint32(0); index < subKeyCount; index++ {
+		accountBufferLength = maximumAccountBufferLength
+		err := windows.RegEnumKeyEx(userNamesKey, index, &accountBuffer[0], &accountBufferLength, nil, nil, nil, nil)
+
+		if err != nil {
+			return "", false, err
+		}
+
+		accountName = windows.UTF16ToString(accountBuffer[:])
+
+		err = changeKeySecurity(userNamesKey, accountName)
+		if err != nil {
+			return "", false, err
+		}
+
+		if accountName == userStr {
+			accountLocated = true
+			break
+		}
+	}
+
+	if accountLocated {
+		var accountKey windows.Handle
+
+		err := windows.RegOpenKeyEx(userNamesKey, &accountBuffer[0], 0, windows.KEY_READ, &accountKey)
+		if err != nil {
+			return "", false, err
+		}
+
+		var userRid uint32
+
+		err = windows.RegQueryValueEx(accountKey, nil, nil, &userRid, nil, nil)
+		if err != nil {
+			return "", false, err
+		}
+
+		accountSid = computerSid + "-" + strconv.FormatUint(uint64(userRid), 10)
+
+		defer windows.RegCloseKey(accountKey)
+	}
+
+	return accountSid, accountLocated, nil
+}
+
+func lookupNTGroup(groupNamesKey windows.Handle, groupStr string, computerSid string) (string, bool, error) {
```

Comment:
This function is identical to `lookupNTUser` except for the variable names. Since it takes the key as a parameter, I'd eliminate this function entirely and rename the variables in `lookupNTUser`: maybe `namesKey`, `nameStr`, and `rid`. Perhaps rename the function to something like `locateNTAccountFromKey`.